vcf.c

```diff
@@ -376,6 +376,96 @@ int bcf_hrec_find_key(bcf_hrec_t *hrec, const char *key)
     return -1;
 }
 
+static int bcf_hrec_set_type(bcf_hrec_t *hrec)
+{
+    if ( !strcmp(hrec->key, "contig") ) hrec->type = BCF_HL_CTG;
+    else if ( !strcmp(hrec->key, "INFO") ) hrec->type = BCF_HL_INFO;
+    else if ( !strcmp(hrec->key, "FILTER") ) hrec->type = BCF_HL_FLT;
+    else if ( !strcmp(hrec->key, "FORMAT") ) hrec->type = BCF_HL_FMT;
+    else if ( hrec->nkeys>0 ) hrec->type = BCF_HL_STR;
+    else hrec->type = BCF_HL_GEN;
+    return 0;
+}
```

Comment:
It probably doesn't matter at all as there aren't many fields to check, but we should probably test the performance impact of this test.

For large if/else-if strcmp lists I tend to use a switch on the first char.  Eg https://github.com/samtools/htslib/blob/develop/sam.c#L1149 (although I see I ought to do the same for the flag name bit).

Ie benchmark it first - likely no change needed.

Reply:
This is reading a VCF header. Note that this code is there elsewhere already, it just has been refactored here. This not a performance concern.